public/app/features/alerting/unified/api/util.ts

```diff
@@ -0,0 +1,17 @@
+import { isFetchError } from '@grafana/runtime';
+
+/**
+ * Catch 404 error response and return "null" instead.
+ *
+ * @example
+ * const ruleGroup = await fetchRuleGroup()
+ *   .unwrap()
+ *   .catch(notFoundToNull); // RuleGroupDTO | null
+ */
+export function notFoundToNull(error: unknown): null {
```

Comment:
I would rename it to `notFoundToNullOrThrow`. Otherwise, it seems not clear to me what the function is actually doing.